src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1286,20 +1286,41 @@ private async ValueTask<(Stream?, TransportContext?, HttpResponseMessage?)> Conn
             }
         }
 
-        private static readonly SocketsConnectionFactory s_defaultConnectionFactory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
+        private static async ValueTask<Stream> DefaultConnectAsync(SocketsHttpConnectionContext context, CancellationToken cancellationToken)
+        {
+            Socket socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
+            socket.NoDelay = true;
+            socket.DualMode = true;
+
+            try
+            {
+                await socket.ConnectAsync(context.DnsEndPoint, cancellationToken).ConfigureAwait(false);
+                return new NetworkStream(socket, true);
+            }
+            catch
+            {
+                socket.Dispose();
+                throw;
+            }
+        }
 
         private ValueTask<Stream> ConnectToTcpHostAsync(string host, int port, HttpRequestMessage initialRequest, bool async, CancellationToken cancellationToken)
         {
             if (async)
             {
-                SocketsConnectionFactory connectionFactory = s_defaultConnectionFactory;
+                Func<SocketsHttpConnectionContext, CancellationToken, ValueTask<Stream>> connectionCallback = Settings._connectionCallback ?? DefaultConnectAsync;
```

Comment:
This is going to allocate a new delegate each time it falls back to the default case. While that is unlikely to break the bank, I'd prefer if it were cached.

Reply:
Will fix